Gradle: Not a supported repository protocol 'artifactregistry' in buildscript block

When using the Gradle plugin for Google Cloud Artifact Registry in the `buildscript` section of a `build.gradle.kts` / `build.gradle` file, the repository protocol `artifactregistry` is not found.

An example repo can be found [here](https://github.com/trietsch/gradle-gcp-artifact-registry). Clone the repository and run `./gradlew clean build`, and you'll be presented with the following error:

```
❯ ./gradlew clean build
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'gradle-artifact-registry'.
> Could not resolve all dependencies for configuration ':classpath'.
   > Not a supported repository protocol 'artifactregistry': valid protocols are [http, https, file, gcs, s3, sftp]
```

Removing the `maven("artifactregistry://...")` from the `buildscript` section works, however, I need to get a `classpath` dependency for the `buildscript` from my Maven Artifact Registry that resides in my GCP project. It seems that the plugin is not applied in the `buildscript` section. I have not found a way to do this, am I doing something wrong?

--- a/artifactregistry-gradle-plugin/src/main/java/com/google/cloud/artifactregistry/gradle/plugin/ArtifactRegistryGradlePlugin.java
+++ b/artifactregistry-gradle-plugin/src/main/java/com/google/cloud/artifactregistry/gradle/plugin/ArtifactRegistryGradlePlugin.java
@@ -32,6 +32,7 @@ import org.gradle.api.artifacts.repositories.ArtifactRepository;
 import org.gradle.api.artifacts.repositories.PasswordCredentials;
 import org.gradle.api.credentials.Credentials;
 import org.gradle.api.initialization.Settings;
+import org.gradle.api.initialization.dsl.ScriptHandler;
 import org.gradle.api.internal.artifacts.repositories.DefaultMavenArtifactRepository;
 import org.gradle.api.invocation.Gradle;
 import org.gradle.api.provider.Property;
@@ -105,6 +106,7 @@ public class ArtifactRegistryGradlePlugin implements Plugin<Object> {
   // The plugin for Gradle will apply CBA repo settings inside settings.gradle and build.gradle.
   private void applyGradle(Gradle gradle, ArtifactRegistryPasswordCredentials crd) {
     gradle.settingsEvaluated(s -> modifySettings(s, crd));
+    gradle.projectsLoaded(g -> g.allprojects(p -> modifyProjectBuildscript(p, crd)));
     gradle.projectsEvaluated(g -> g.allprojects(p -> modifyProject(p, crd)));
   }
 
@@ -123,6 +125,15 @@ public class ArtifactRegistryGradlePlugin implements Plugin<Object> {
     final PublishingExtension publishingExtension = p.getExtensions().findByType(PublishingExtension.class);
     if (publishingExtension != null) {
       publishingExtension.getRepositories().forEach(r -> configureArtifactRegistryRepository(r, crd));
+    }
+  }
+
+  // Not sure this knows which repositories already exist, so register a callback to modify the repos
+  // as they are added to the buildscript
+  private void modifyProjectBuildscript(Project p, ArtifactRegistryPasswordCredentials crd) {
+    final ScriptHandler buildscript = p.getBuildscript();
+    if (buildscript != null) {
+      buildscript.getRepositories().whenObjectAdded(r -> configureArtifactRegistryRepository(r, crd));
     }
   }
 

